js/engine/Main.ml

```diff
@@ -24,20 +37,36 @@ let _ =
 
        method deleteFile filename = Sys.remove (Js.to_string filename)
 
-       method setParsePattern (func : bool -> Lang.t -> string -> 'a) =
-         Parse_pattern.parse_pattern_ref := func
+       method setParsePattern
+           (func : bool -> Js.js_string Js.t -> Js.js_string Js.t -> 'a) =
+         Parse_pattern.parse_pattern_ref :=
+           fun print_error lang target ->
+             func print_error
+               (lang |> Lang.to_lowercase_alnum |> String.lowercase_ascii
```

Comment:
Forgot to add a fallout for this: I noticed that the first character of to_lowercase_alnum is uppercase (caught this in the js tests), and I think it just hasn’t been a problem yet because Lang.of_string is case insensitive. We should probably fix though

Reply:
For now, I'm going to remove the `String.lowercase_ascii`s and normalize the parser name when the FE fetches the js/wasm

Reply:
There are a few callers in Test_parsing.ml in semgrep and in Deep_actions.ml in semgrep-pro, but maybe the code can be rewritten.

Reply:
factored out the fix here: https://github.com/returntocorp/semgrep/pull/7761